contrib/packs/actions/pack_mgmt/download.py

```diff
@@ -44,10 +62,23 @@ def __init__(self, config=None):
         self._repo_url = None
 
     def run(self, packs, repo_url, abs_repo_base, verifyssl=True, branch='master', subtree=False):
+
+        cached_repo_url, cached_branch, cached_subtree = self._lookup_cached_gitinfo(
+            abs_repo_base, packs)
+
+        if not repo_url:
+            repo_url = cached_repo_url
+        if not branch:
```

Comment:
Using the aliases from https://github.com/StackStorm/st2/blob/master/contrib/packs/aliases/pack_deploy.yaml#L6 as reference 

`pack deploy {{packs}} {{branch=''}} {{repo_url=''}}` should be suppportable